dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/LogTraceData.java

```diff
@@ -93,22 +93,22 @@ private void setPair(int index, long l) {
     }
 
     private String getStringValue(int index) {
-        KeyValueStringPair kvp = (KeyValueStringPair) getPairs().get(index);
-        return kvp.getStringValue();
+        KeyValueStringPair kvp = (KeyValueStringPair) getPairs()[index];
+        return kvp == null ? null : kvp.getStringValue();
```

Comment:
to be consistent with getIntValue / getLongValue, we should not check for nulls here and just let an NPE happen in cases where a kvp doesn't exist.  That way anyone using this class can rely on NPEs as the way of indicating a field is missing.

Reply:
I changed this to handle null because there were a few fields in CollectorJSONUtils that were calling get<FieldName> on some string KVPs that were running into NPEs. For example, getMessageId(), getThrowable(), and getSequence()

Reply:
This is a case where Java's Optional class could help, but I'm hesitant to introduce that as it would result in a lot of extra method calls for this use case.  Instead, could we add a second getStringValue that takes two parms, the second indicating the value to return if the key isn't present?  If we do that, should provide same for int/long for consistency.

```
public String getStringValue(int index, String default) { ... }
```

Reply:
As per our discussion, I'll create a new issue for this change and leave my changes as is.